WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/Create Shipping Label Form/Package Details/ShippingLabelPackageDetailsViewModel.swift

```diff
@@ -1,50 +1,148 @@
+
+    init(order: Order,
+         formatter: CurrencyFormatter = CurrencyFormatter(currencySettings: ServiceLocator.currencySettings),
+         stores: StoresManager = ServiceLocator.stores,
+         storageManager: StorageManagerType = ServiceLocator.storageManager,
+         weightUnit: String? = ServiceLocator.shippingSettingsService.weightUnit) {
+        self.order = order
+        self.orderItems = order.items
+        self.currency = order.currency
+        self.currencyFormatter = formatter
+        self.stores = stores
+        self.storageManager = storageManager
+        self.weightUnit = weightUnit
+        configureResultsControllers()
+        syncProducts()
+        syncProductVariations()
+    }
+
+    private func configureResultsControllers() {
+        resultsControllers = ShippingLabelPackageDetailsResultsControllers(siteID: order.siteID,
+                                                                           orderItems: order.items,
+                                                                           storageManager: storageManager,
+           onProductReload: { [weak self] (products) in
+            guard let self = self else { return }
+            self.products = products
+            self.itemsRows = self.generateItemsRows()
+        }, onProductVariationsReload: { [weak self] (productVariations) in
+            guard let self = self else { return }
+            self.productVariations = productVariations
+            self.itemsRows = self.generateItemsRows()
+        })
+
+        products = resultsControllers?.products ?? []
+        productVariations = resultsControllers?.productVariations ?? []
+        itemsRows = generateItemsRows()
+    }
+
+    /// Generate the items rows, creating an element in the array for every item (eg. if there is an item with quantity 3,
+    /// we will generate 3 different items), and we will remove virtual products. 
+    ///
+    private func generateItemsRows() -> [ItemToFulfillRow] {
+        var itemsToFulfill: [ItemToFulfillRow] = []
+        for item in orderItems {
+            let isVariation = item.variationID > 0
+            var product: Product?
+            var productVariation: ProductVariation?
+
+            if isVariation {
+                productVariation = productVariations.first { $0.productVariationID == item.variationID }
+            }
+            else {
+                product = products.first { $0.productID == item.productID }
+            }
+            if product?.virtual == false || productVariation?.virtual == false {
+                for _ in 0..<item.quantity.intValue {
+                    let attributes = item.attributes.map { VariationAttributeViewModel(orderItemAttribute: $0) }
+                    let weightUnitA: String = weightUnit ?? ""
+                    let subtitle = Localization.subtitle(weight: isVariation ? productVariation?.weight : product?.weight, attributes: attributes) + weightUnitA
```

Comment:
The weight unit should probably be part of the localization formatter too, not appended afterwards, especially for RTL languages.

I'm not sure they change from language to language, but core localizes them too: https://github.com/woocommerce/woocommerce/blob/b446417be69602ffce3d9023bd2b3eef39318451/includes/admin/settings/class-wc-settings-products.php#L363-L366

Reply:
That said, I also think there should be a space between the weight and the unit, e.g. `1 lb`

Maybe we'd benefit from a formatter for weight like we have for [prices](https://github.com/woocommerce/woocommerce-ios/blob/develop/WooCommerce/Classes/Tools/Currency/CurrencyFormatter.swift).